firmware/src/app.c

```diff
@@ -114,12 +115,14 @@ static void APP_ConnectNotifyCb(DRV_HANDLE handle, WDRV_WINC_CONN_STATE currentS
 // Section: Global Data Definitions
 // *****************************************************************************
 // *****************************************************************************
+#define SN_STRING "sn"
 
 /* Place holder for ECC608A unique serial id */
 ATCA_STATUS appCryptoClientSerialNumber;
 ATCA_STATUS retValCryptoClientSerialNumber;
 char* attDeviceID;
-char attDeviceID_buf[20] = "BAAAAADD1DBAAADD1D";
+char attDeviceID_buf[25] = "BAAAAADD1DBAAADD1D";
+char buf[25];
```

Comment:
This is not big enough, you also copy "sn" into it. line 415

Reply:
you should not call this buf, because it is assigned to attDeviceID

Reply:
The serial number is 18 characters long, including "sn" should be ok with 25.